mail_tracking/static/src/js/failed_message/discuss.js

```diff
@@ -0,0 +1,398 @@
+            'click .o_failed_message_retry': '_onRetryFailedMessage',
+            'click .o_failed_message_reviewed': '_onMarkFailedMessageReviewed',
+        }),
+
+        /**
+        * Paramaters used to render 'failed' mailbox entry in Discuss
+        *
+        * @private
+        * @returns {Object}
+        */
+        _sidebarQWebParams: function () {
+            var failed = this.call('mail_service', 'getMailbox', 'failed');
+            return {
+                activeThreadID: this._thread ? this._thread.getID() : undefined,
+                failedCounter: failed.getMailboxCounter(),
+            };
+        },
+
+        /**
+        * Render 'failed' mailbox menu entry in Discuss
+        *
+        * @private
+        * @returns {jQueryElementt}
+        */
+        _renderSidebar: function () {
+            var $sidebar = this._super.apply(this, arguments);
+            // Because Odoo implementation isn't designed to be inherited
+            // properly, we inject 'failed' button using jQuery.
+            var $failed_item = $(QWeb.render('mail_tracking.SidebarFailed',
+                this._sidebarQWebParams()));
+            $failed_item.insertAfter(
+                $sidebar.find(".o_mail_discuss_title_main").filter(":last"));
+            return $sidebar;
+        },
+
+        /**
+         * Overrides to listen click on 'Set all as reviewed' button
+         *
+         * @Override
+         */
+        _renderButtons: function () {
+            this._super.apply(this, arguments);
+            this.$buttons
+                .on('click', '.o_mail_discuss_button_set_all_reviewed',
+                    this._onSetAllAsReviewedClicked.bind(this));
+        },
+
+        /**
+         * Overrides to update 'set all as reviewed' button
+         *
+         * @Override
+         */
+        _updateControlPanelButtons: function (thread) {
+            // Set All Reviewed
+            if (thread.getID() === 'mailbox_failed') {
+                this.$buttons
+                    .find('.o_mail_discuss_button_set_all_reviewed')
+                    .removeClass('d-none d-md-inline-block')
+                    .addClass('d-none d-md-inline-block');
+            } else {
+                this.$buttons
+                    .find('.o_mail_discuss_button_set_all_reviewed')
+                    .removeClass('d-none d-md-inline-block')
+                    .addClass('d-none');
+            }
+
+            this._super.apply(this, arguments);
+        },
+
+        /**
+         * Overrides to update 'set all as reviewed' button
+         *
+         * @Override
+         */
+        _updateButtonStatus: function (disabled, type) {
+            if (this._thread.getID() === 'mailbox_failed') {
+                this.$buttons
+                    .find('.o_mail_discuss_button_set_all_reviewed')
+                    .toggleClass('disabled', disabled);
+                //  Display Rainbowman when all inbox messages are reviewed
```

Comment:
```suggestion
                // Display Rainbowman when all failed messages are reviewed
```


BTW This idea is awesome. :raised_hands: